client/state/selectors/get-current-user-payment-methods.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { kebabCase, upperCase } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentUserLocale } from 'state/current-user/selectors';
+import { getGeoCountryShort } from 'state/geo/selectors';
+
+/**
+ * Constants
+ *
+ * Payment methods can be targeted via a WP.com locale code (actually
+ * a two letter lang usually) or by a full lang-CC locale.
+ */
+const default_payment_methods = [ 'credit-card', 'paypal' ];
+
+const paymentMethodsByLocale = {
```

Comment:
So... are you more in favor of something like https://github.com/Automattic/wp-calypso/pull/12827 then?

I think we might have a general i18n meta endpoint at some point in the future--of which payment prefs by locale would be a subset of the overall data. But, it could be awhile before we would have much need for something like that. The near future looks like a small amount of targeted experiments to me.